app/src/main/java/org/mozilla/fenix/utils/Settings.kt

```diff
@@ -0,0 +1,48 @@
+package org.mozilla.fenix.utils
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.preference.PreferenceManager
+import org.mozilla.fenix.R
+import org.mozilla.fenix.ext.getPreferenceKey
+
+/**
+ * A simple wrapper for SharedPreferences that makes reading preference a little bit easier.
+ */
+class Settings private constructor(context: Context) {
+
+    companion object {
+        var instance: Settings? = null
+
+        @JvmStatic
+        @Synchronized
+        fun getInstance(context: Context): Settings {
+            if (instance == null) {
+                instance = Settings(context.applicationContext)
+            }
+            return instance ?: throw AssertionError("Instance cleared")
+        }
+    }
+
+    private val appContext = context.applicationContext
+
+    private val preferences: SharedPreferences =
+        PreferenceManager.getDefaultSharedPreferences(context)
+
+    val defaultSearchEngineName: String
```

Comment:
Do we still need the `!!` in here since we're giving it a default? And if we do let's switch it to `?:`

Reply:
tru